### ISSUE TITLE
String.to_integer("+") returns 0

### ISSUE THREAD

**Issue by niku** [elixir-lang/elixir#3538]
`String.to_integer("+")` and `String.to_integer("-")` return 0.
There are not expected behevior.

Related https://groups.google.com/forum/#!topic/elixir-lang-talk/U-cKQyWqwTY

```
/Users/niku% iex
Erlang/OTP 17 [erts-6.4] [source] [64-bit] [smp:4:4]
[async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Interactive Elixir (1.0.5) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> String.to_integer("42")
42
iex(2)> String.to_integer("a")
** (ArgumentError) argument error
    :erlang.binary_to_integer("a") 
iex(2)> String.to_integer("+")
0
iex(3)> String.to_integer("-")
0 
```

**Comment by josevalim** [comment]
Great find, thank you!

**Comment by niku** [comment]
I'll try to fix it.

But I have a question.
Which branch should I check out to send a pull-request `master` or `v1.0`?

**Comment by jduan** [comment]
String.to_integer directly invokes :erlang.binary_to_integer. So is this a bug in Erlang itself?

iex(1)> :erlang.binary_to_integer("+")
0
iex(2)> :erlang.binary_to_integer("-")
0

**Comment by josevalim** [comment]
Ah, yes. It is an Erlang bug. :S Can someone please report it to Erlang's tracker?

**Comment by niku** [comment]
OK. I'll send a report.

**Comment by josevalim** [comment]
Thank you! You'll need to send Erlang snippets though!

```
$ erl
Erlang/OTP 17 [erts-6.4] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Eshell V6.4  (abort with ^G)
1> binary_to_integer(<<"+">>).
0
2> binary_to_integer(<<"-">>).
0
```

**Comment by niku** [comment]
Fixed by erlang/otp#803 in Erlang/OTP 18.1 

Thanks.

```
/Users/niku% erl
Erlang/OTP 18 [erts-7.1] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Eshell V7.1  (abort with ^G)
1> code:root_dir().
"/usr/local/Cellar/erlang/18.1/lib/erlang"
2> binary_to_integer(<<"+">>).
** exception error: bad argument
     in function  binary_to_integer/1
        called as binary_to_integer(<<"+">>)
3> binary_to_integer(<<"-">>).
** exception error: bad argument
     in function  binary_to_integer/1
        called as binary_to_integer(<<"-">>)

/Users/niku% iex
Erlang/OTP 18 [erts-7.1] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Interactive Elixir (1.0.5) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> String.to_integer("+")
** (ArgumentError) argument error
    :erlang.binary_to_integer("+")
iex(1)> String.to_integer("-")
** (ArgumentError) argument error
    :erlang.binary_to_integer("-")
```
